bundles/org.openhab.binding.epsonprojector/src/main/java/org/openhab/binding/epsonprojector/internal/EpsonProjectorHandler.java

```diff
@@ -0,0 +1,451 @@
+                    return new DecimalType(lampTime);
+                case LUMINANCE:
+                    Luminance luminance = remoteController.getLuminance();
+                    return new StringType(luminance.toString());
+                case MUTE:
+                    Switch mute = remoteController.getMute();
+                    return mute == Switch.ON ? OnOffType.ON : OnOffType.OFF;
+                case OFFSET_BLUE:
+                    int offsetBlue = remoteController.getOffsetBlue();
+                    return new DecimalType(offsetBlue);
+                case OFFSET_GREEN:
+                    int offsetGreen = remoteController.getOffsetGreen();
+                    return new DecimalType(offsetGreen);
+                case OFFSET_RED:
+                    int offsetRed = remoteController.getOffsetRed();
+                    return new DecimalType(offsetRed);
+                case POWER:
+                    PowerStatus powerStatus = remoteController.getPowerStatus();
+
+                    if (powerStatus == PowerStatus.ON) {
+                        return OnOffType.ON;
+                    } else {
+                        return OnOffType.OFF;
+                    }
+                case POWER_STATE:
+                    PowerStatus powerStatus1 = remoteController.getPowerStatus();
+                    return new StringType(powerStatus1.toString());
+                case SHARP:
+                    logger.warn("Get '{}' not implemented!", commmandType.toString());
+                    return UnDefType.UNDEF;
+                case SOURCE:
+                    Source source = remoteController.getSource();
+                    return new StringType(source.toString());
+                case SYNC:
+                    int sync = remoteController.getSync();
+                    return new DecimalType(sync);
+                case TINT:
+                    int tint = remoteController.getTint();
+                    return new DecimalType(tint);
+                case TRACKING:
+                    int tracking = remoteController.getTracking();
+                    return new DecimalType(tracking);
+                case VKEYSTONE:
+                    int vKeystone = remoteController.getVerticalKeystone();
+                    return new DecimalType(vKeystone);
+                case VPOSITION:
+                    int vPosition = remoteController.getVerticalPosition();
+                    return new DecimalType(vPosition);
+                case VREVERSE:
+                    Switch vReverse = remoteController.getVerticalReverse();
+                    return vReverse == Switch.ON ? OnOffType.ON : OnOffType.OFF;
+                default:
+                    logger.warn("Unknown '{}' command!", commmandType);
+                    return UnDefType.UNDEF;
+            }
+
+        } catch (EpsonProjectorException e) {
+            logger.debug("Couldn't execute command '{}', {}", commmandType.toString(), e.getMessage());
```

Comment:
You don't have to call `toString` it will be called automatically, but only when the logging-level actually applies.


Please check all